### ISSUE TITLE
Adding @netlify prefix broke the esbuild no-optional workaround 

### ISSUE THREAD

**Issue by pimlie** [eduardoboucas/esbuild#17]
See the below build output. Adding the @netlify prefix seems to have broken the download path used by esbuild. 

It should be
`https://registry.npmjs.org/@netlify/esbuild-linux-64/-/esbuild-linux-64-0.14.23.tgz`
and not 
`https://registry.npmjs.org/@netlify/esbuild-linux-64/-/@netlify/esbuild-linux-64-0.14.23.tgz`

```
12:47:44 PM: error /opt/build/repo/node_modules/@netlify/esbuild: Command failed.
12:47:44 PM: Exit code: 1
12:47:44 PM: Command: node install.js
12:47:44 PM: Arguments:
12:47:44 PM: Directory: /opt/build/repo/node_modules/@netlify/esbuild
12:47:44 PM: Output:
12:47:44 PM: [esbuild] Failed to find package "@netlify/esbuild-linux-64" on the file system
12:47:44 PM: This can happen if you use the "--no-optional" flag. The "optionalDependencies"
12:47:44 PM: package.json feature is used by esbuild to install the correct binary executable
12:47:44 PM: for your current platform. This install script will now attempt to work around
12:47:44 PM: this. If that fails, you need to remove the "--no-optional" flag to use esbuild.
12:47:44 PM: [esbuild] Trying to install package "@netlify/esbuild-linux-64" using npm
12:47:44 PM: [esbuild] Failed to install package "@netlify/esbuild-linux-64" using npm: ENOENT: no such file or directory, rename '/opt/build/repo/node_modules/esbuild/lib/npm-install/node_modules/@netlify/esbuild-linux-64/bin/esbuild' -> '/opt/build/repo/node_modules/esbuild/lib/downloaded-@netlify/esbuild-linux-64-esbuild'
12:47:44 PM: [esbuild] Trying to download "https://registry.npmjs.org/@netlify/esbuild-linux-64/-/@netlify/esbuild-linux-64-0.14.23.tgz"
12:47:44 PM: [esbuild] Failed to download "https://registry.npmjs.org/@netlify/esbuild-linux-64/-/@netlify/esbuild-linux-64-0.14.23.tgz": Server responded with 404
12:47:44 PM: /opt/build/repo/node_modules/@netlify/esbuild/install.js:236
12:47:44 PM:         throw new Error(`Failed to install package "${pkg}"`);
12:47:44 PM:               ^
12:47:44 PM: Error: Failed to install package "@netlify/esbuild-linux-64"
12:47:44 PM:     at checkAndPreparePackage (/opt/build/repo/node_modules/@netlify/esbuild/install.js:236:15)
12:47:44 PM:     at processTicksAndRejections (node:internal/process/task_queues:96:5)
12:47:44 PM: info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
12:47:51 PM: Error during Yarn install
12:47:51 PM: Build was terminated: Build script returned non-zero exit code: 1
12:47:51 PM: Creating deploy upload records
```

**Comment by Skn0tt** [comment]
Hi! Can you add some reproductions steps? Running `yarn add @netlify/esbuild` works fine for me, but i'm on `darwin-arm64`. I assume you're running this on Linux?

**Comment by pimlie** [comment]
Sorry, I missed a step in my report!

Locally everything works fine because I install the dependencies with `yarn install`

But in the Deploy Settings on Netlify I have set:
```
  YARN_FLAGS = "--frozen-lockfile --production --ignore-optional"
```

and then the build fails due to `--ignore-optional`

As workaround I have removed ignore-optional for now but this means that we are installing more dependencies then we need so build time could take a bit longer :)

**Comment by Skn0tt** [comment]
Interesting! Is `@netlify/esbuild` part of *your* custom build process, or does it fail as a transitive dependency of https://github.com/netlify/build?

**Comment by pimlie** [comment]
Using `yarn why` we are using it because we use `@netlify/zip-it-and-ship-it` (which has a dependency on netlify/build indeed)

**Comment by Skn0tt** [comment]
Okay, that's good info! Thank you :) I'll take some time next week to look into this.

**Comment by Skn0tt** [comment]
okay, faster than expected ^^ I'll release this together with #16 once that's merged, will ping back to let you know.

**Comment by pimlie** [comment]
Awesome, thats super quick indeed. Thanks! 🎉

**Comment by Skn0tt** [comment]
Published in [v0.14.25](https://www.npmjs.com/package/@netlify/esbuild/v/0.14.25). Can you check if that fixes it for you?
